File: src/Entity/Application.php
<?php

namespace App\Entity;

use App\Repository\ApplicationRepository;
use Doctrine\DBAL\Types\Types;
use Doctrine\ORM\Mapping as ORM;
use Symfony\Component\Validator\Constraints as Assert;

#[ORM\Entity(repositoryClass: ApplicationRepository::class)]
class Application
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column]
    private ?int $id = null;

    #[ORM\Column(type: Types::DATE_MUTABLE ,name: "dateDebut")]
    #[Assert\NotNull(message: "La date de début est obligatoire.")]
    #[Assert\GreaterThanOrEqual("today", message: "La date de début ne peut pas être dans le passé.")]
    #[Assert\Type("\DateTimeInterface")]
    private ?\DateTimeInterface $dateDebut = null;

    #[ORM\Column(type: Types::DATE_MUTABLE  ,name: "dateFin")]
    #[Assert\NotNull(message: "La date de fin est obligatoire.")]
    #[Assert\Type("\DateTimeInterface")]
    #[Assert\GreaterThan(propertyPath: "dateDebut", message: "La date de fin doit être postérieure à la date de début.")]
    private ?\DateTimeInterface $dateFin = null;

    #[ORM\ManyToOne(inversedBy: 'applications')]
    #[ORM\JoinColumn(nullable: false , name: 'idProgramme', referencedColumnName: 'id',onDelete: 'CASCADE')]
    #[Assert\NotNull(message: "Le programme est requis.")]
    private ?Programme $idProgramme = null;

    #[ORM\ManyToOne(inversedBy: 'applications')]
    #[ORM\JoinColumn(nullable: false , name: 'idClient', referencedColumnName: 'id',onDelete: 'CASCADE')]
    #[Assert\NotNull(message: "Le client est requis.")]
    private ?Client $idClient = null;

    public function getId(): ?int
    {
        return $this->id;
    }

    public function getDateDebut(): ?\DateTimeInterface
    {
        return $this->dateDebut;
    }

    public function setDateDebut(?\DateTimeInterface $dateDebut): static
    {
        $this->dateDebut = $dateDebut;

        return $this;
    }

    public function getDateFin(): ?\DateTimeInterface
    {
        return $this->dateFin;
    }

    public function setDateFin(?\DateTimeInterface $dateFin): static
    {
        $this->dateFin = $dateFin;

        return $this;
    }

    public function getIdProgramme(): ?Programme
    {
        return $this->idProgramme;
    }

    public function setIdProgramme(?Programme $idProgramme): static
    {
        $this->idProgramme = $idProgramme;

        return $this;
    }

    public function getIdClient(): ?Client
    {
        return $this->idClient;
    }

    public function setIdClient(?Client $idClient): static
    {
        $this->idClient = $idClient;

        return $this;
    }
}

File: src/main/java/com/esprit/models/application.java
package com.esprit.models;

import java.time.LocalDate;

public class application {

    private int id;
    private LocalDate dateDebut;
    private LocalDate dateFin;
    private int idProgramme;
    private int idClient;

    // ✅ Constructeur
    public application( int id,LocalDate dateDebut, LocalDate dateFin, int idProgramme, int idClient) {
        this.id=id;

        this.dateDebut = dateDebut;
        this.dateFin = dateFin;
        this.idProgramme = idProgramme;
        this.idClient = idClient;
    }
    // 🔹 Constructeur avec 4 paramètres (sans ID)
    public application(LocalDate dateDebut, LocalDate dateFin, int idProgramme, int idClient) {
        this.dateDebut = dateDebut;
        this.dateFin = dateFin;
        this.idProgramme = idProgramme;
        this.idClient = idClient;
    }

    // Getters et Setters
    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public LocalDate getDateDebut() {
        return dateDebut;
    }

    public void setDateDebut(LocalDate dateDebut) {
        this.dateDebut = dateDebut;
    }

    public LocalDate getDateFin() {
        return dateFin;
    }

    public void setDateFin(LocalDate dateFin) {
        this.dateFin = dateFin;
    }

    public int getIdProgramme() {
        return idProgramme;
    }

    public void setIdProgramme(int idProgramme) {
        this.idProgramme = idProgramme;
    }

    public int getIdClient() {
        return idClient;
    }

    public void setIdClient(int idClient) {
        this.idClient = idClient;
    }

    @Override
    public String toString() {
        return "Application{" +
                "id=" + id +
                ", dateDebut=" + dateDebut +
                ", dateFin=" + dateFin +
                ", idProgramme=" + idProgramme +
                ", idClient=" + idClient +
                '}';
    }
}
